wordpress-mobile/gutenberg-mobile
# 1.29.0 Release

Release for Gutenberg Mobile v1.29.0

## Related PRs

- Gutenberg: No changes on [`rnmobile/release-1.29.0`](https://github.com/WordPress/gutenberg/tree/rnmobile/release-1.29.0) for a PR yet
- WPAndroid: https://github.com/wordpress-mobile/WordPress-Android/pull/12033
- WPiOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/14192
- Aztec-iOS: N/A
- Aztec-Android: N/A

## Additional Changes that Landed After the Release Was Cut

- Fix for editing links, that moves the cursor/caret to after the link: https://github.com/WordPress/gutenberg/pull/22652
- Fix for crash on delete of the Media-Text and Contact Info blocks: https://github.com/WordPress/gutenberg/pull/22677
- Add freeform support to the react native editor: https://github.com/WordPress/gutenberg/pull/22609
- 1.28.2 hotfix, Only release Pullquote block on iOS: https://github.com/WordPress/gutenberg/commit/b96445f4c00caff30dfbff2eb897e32f9d54f86f
- Correct button block merge: https://github.com/WordPress/gutenberg/pull/22708#pullrequestreview-420550206
- Fix footer visibility on buttons blocks when deselecting: https://github.com/WordPress/gutenberg/pull/22711
- Fix removal of buttons when removing all buttons: https://github.com/WordPress/gutenberg/pull/22716
- Fix Color updates when changing between solid and gradient colors: https://github.com/WordPress/gutenberg/pull/22736
- Fix: Add default value to format when start and end are undefined: https://github.com/WordPress/gutenberg/pull/22741

### Incoming

- Button Block Fixes
  - Correct concatenation in Button block merge function: https://github.com/WordPress/gutenberg/pull/22708
  - Fix footer appender in buttons block: https://github.com/WordPress/gutenberg/pull/22711
  - Add onDeleteBlock to the extra data to re-render when it has changes: https://github.com/WordPress/gutenberg/pull/22716

## Changes


- Add support for changing overlay color settings in Cover block: https://github.com/WordPress/gutenberg/pull/22207
- Entry/exit animation in FloatingToolbar: https://github.com/WordPress/gutenberg/pull/22146
- Verse Block: https://github.com/WordPress/gutenberg/pull/21883
- Fix merged of text blocks when a style is active: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2271
- Block toolbar can now collapse when the block width is smaller than the toolbar content: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2220
- Trash icon that is used to remove blocks is moved to the new menu reachable via ellipsis button in the block toolbar: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2220
- Creating undo levels less frequently: https://github.com/WordPress/gutenberg/pull/22492
- Tooltip for page template selection buttons: https://github.com/WordPress/gutenberg/pull/21974
- Fix button alignment in page templates and make strings consistent: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2314
- Add support for displaying radial gradients in Buttons and Cover blocks: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2266

## Test plan

1. Use the main WP apps to test the changes above. 
2. Check WPAndroid and WPiOS PRs if there are specific tests to run.
3. Smoke test the main WP apps for [general writing flow](https://github.com/wordpress-mobile/test-cases/tree/master/test-cases/gutenberg/writing-flow).
4. Verify that the release notes on the WP app PRs have been correctly updated.

## Release Submission Checklist

  - iOS: 'grep WordPressAztec-iOS RNTAztecView.podspec'
  - Android: 'grep aztecVersion react-native-aztec/android/build.gradle'
